wandb/sdk/artifacts/artifacts_cache.py

```diff
@@ -181,6 +181,13 @@ def helper(mode: str = "w") -> Generator[IO, None, None]:
             try:
                 yield temp_file
                 temp_file.close()
+                # The NamedTemporaryFile is always created with mode 0600, hardcoded at
+                # tempfile.py, line 235, because it is private to your process until
+                # you open it up with chmod. Here we change the permissions back to
+                # the system defaults, with a temporary umask assignment to 0o022
+                umask = os.umask(0o22)
```

Comment:
This is not thread safe, see: https://stackoverflow.com/a/44130549

Reply:
I think this is the only safe way to get the umask: https://stackoverflow.com/a/53357635

Reply:
updated